src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -501,12 +501,20 @@ def astype(self, dtype: Any, copy: bool = True) -> Index:
         >>> idx.astype('float')
         Index([1.0, 2.0, 3.0], dtype='float64')
         """
-        WarningMessage.index_to_pandas_warning("astype")
-        return Index(
-            self.to_pandas().astype(dtype=dtype, copy=copy),
-            dtype=dtype,
-            convert_to_lazy=self.is_lazy,
-        )
+        if dtype is not None:
+            dtype = pandas_dtype(dtype)
+
+        if self.dtype == dtype:
+            # Ensure that self.astype(self.dtype) is self
+            return self.copy() if copy else self
+
+        col_dtypes = {
+            column: pandas_dtype(dtype)
+            for column in self._query_compiler.get_index_names()
+        }
+        new_query_compiler = self._query_compiler.astype(col_dtypes, is_index=True)
+        self.set_query_compiler(new_query_compiler)
```

Comment:
is astype an inplace operations? or it returns a new object? self.set_query_compiler sounds like we are replacing the query compiler of the current object? However, if we are returning a new object, we should create a new Index class object here, not self?

Reply:
Good catch, I fixed that. This is not an inplace operation.